lib/carlo.js

```diff
@@ -471,7 +471,9 @@ async function launch(options = {}) {
   if (!options.bgcolor)
     options.bgcolor = '#ffffff';
 
-  const executablePath = options.executablePath || findChrome().pop();
+  let executablePath = options.executablePath || findChrome().pop();
+  if (!executablePath && options.downloadChromiumIfNeeded)
```

Comment:
I wonder if we should allow following options:
`['stable']`, <-- the default
`['canary']`, <-- if I only want to use Canary
`['stable', 'canary', 'chromium']` <-- If I want stable, canary or latest chromium (downloaded)
`['*']` <-- don't care, just make it work
`['r608001']` <-- I know that I need particular chromium revision

You don't need to do it, but establishing the format under `options.configurations = ['chromium']` right away would be a good start. I know where to get the code for Canary as well.

Reply:
I concur with @pavelfeldman. This seems like a great solution, especially if you depend on a specific chromium version.

Reply:
I think that if carlo is aiming to provide secure and updatable environment then `'r608001'` should be hard to use option..
What is about following configuration? I can specify list of local builds that my app should use ordered by priority and / or one latest channel that will be downloaded if local is not available.
```javascript
{
  local: ['stable', 'canary'] // optional list of local chrome channels to lookup
  latest: 'chromium' // optional fallback channel, latest will be downloaded
  downloadAfter: 7 days // optional interval since last download to avoid chromium downloading on each app start
  versionAvailableCallback function // optional callback that app can specify to warn user about new version
}
```

If I would like to use only local stable: `{local: 'stable'}`.
If I would like to use local stable or canary: `{local: ['stable', 'canary']}`.
If I would like to download latest canary and later use it: `{local: 'canary', latest: 'canary'}`
If I would like to use canary that is not older then 5 days: `{latest: 'canary', downloadAfter: '5 days'}`.
If I would like to be brave and use latest chromium: `{latest: 'chromium'}`.

Reply:
We can't really 'load' canary or stable, we can only suggest whether we work with them or not. So the options really are either local stable, local canary or _some_ chromium. I'm hesitant calling it `latest`, because it is really up to puppeteer which version is used. My `rXXXXX` option is an escape hatch for power power power users, I expect it to never be used, but I don't want to make it artificially inconvenient because it is orthogonal to security, you are at your own risk with chromium anyways.

Reply:
I added support for ['*'], ['chromium'] and ['r599821'], please take another look.

Reply:
and ['stable'] actually that I would call 'local' instead and convert configurations array to configuration string: 'local' or 'chromium' or 'r...' or '*'.